src/main/java/htsjdk/tribble/readers/BlockCompressedAsciiLineReader.java

```diff
@@ -0,0 +1,42 @@
+package htsjdk.tribble.readers;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.samtools.util.CloserUtil;
+
+import java.io.IOException;
+
+/**
+ * An AsciiLineReader implementation that wraps a BlockCompressedInputStream and provides no additional buffering.
+ * Useful for cases where we need to preserve virtual file pointers in the underlying stream, such as during indexing.
+ */
+class BlockCompressedAsciiLineReader extends AsciiLineReader {
+
+    final private BlockCompressedInputStream bcs;
+
+    public BlockCompressedAsciiLineReader(final BlockCompressedInputStream inputBlockCompressedStream) {
+        bcs = inputBlockCompressedStream;
+    }
+
+    /**
+     * Read a single line of input, advance the underlying stream only enough to read the line.
+     */
+    public String readLine() throws IOException {
```

Comment:
If these are overridden, they need an `@Override` annotation. Also there's no need to javadoc an overridden method, as javadoc will copy down the base class method doc.